lib/blob.js

```diff
@@ -359,19 +604,496 @@ Blob.prototype.getContainerMetadata = function getContainerMetadata(name) {
+ *          eTag: '...',                  // The entity tag of the blob
+ *          contentLength: '...',         // The content length of the blob
+ *          contentType: '...',           // The MIME content type of the blob
+ *          contentEncoding: '...',       // The content encoding of the blob
+ *          contentLanguage: '...',       // The content language of the blob
+ *          contentMD5: '...',            // An MD5 hash of the blob content
+ *          cacheControl: '...',          // The blob cache control
+ *          xmsBlobSequenceNumber: '...', // The page blob sequence number
+ *          blobType: '...',              // The type of the blob: BlockBlob | PageBlob | AppendBlob
+ *          leaseStatus: '...',           // The lease status of the blob
+ *          leaseState: '...',            // The lease state of the blob
+ *          leaseDuration: '...',         // The lease duration of the blob
+ *          copyId: '...',                // String identifier for the copy operation
+ *          copyStatus: '...',            // The state of the copy operation: pending | success | aborted | failed
+ *          copySource: '...',            // The name of the source blob of the copy operation
+ *          copyProgress: '...',          // The bytes copied/total bytes
+ *          copyCompletionTime: '...',    // The date and time the copy operation finished
+ *          copyStatusDescription: '...', // The status of the copy operation
+ *          serverEncrypted: false,       // true if the blob and application metadata are completely encrypted, and false otherwise
+ *          incrementalCopy: '...',       // true for the incremental copy blobs operation and snapshots
+ *       }
+ *       metadata:   {}                   // Meta-data dictionary if requested
+ *     }
+ *   ],
+ *   blobPrefixName: '...',
+ *   prefix:         '...',               // prefix given in options (if given)
+ *   marker:         '...',               // marker given in options (if given)
+ *   maxResults:     5000,                // maxResults given in options (if given)
+ *   nextMarker:     '...'                // Next marker if not at end of list
+ *   delimiter:      '...'                // Delimiter
+ * }
+ * ```
+ */
+Blob.prototype.listBlobs = function listBlobs(container, options) {
+  assert(container, 'The name of the container must be specified');
+  // Construct the query string
+  var query = {
+    restype: 'container',
+    comp: 'list'
+  };
+
+  assert(container, 'The container name must be specified');
+
+  if (options) {
+    if (options.prefix)     query.prefix      = options.prefix;
+    if (options.marker)     query.marker      = options.marker;
+    if (options.maxResults) query.maxresults  = options.maxResults;
+    if (options.include)  {
+      var includeValue = '';
+      if (options.include.snapshot) {
+        includeValue += 'snapshot%82';
```

Comment:
is there a reason for using %82 and not building the string using the real character and then encoding into the final string form?

Another pattern we've used elsewhere is to build up a list of strings then run the .join(X) function on them, whic can sometimes read a little cleaner.